### PR TITLE
Addon customization story

### PR DESCRIPTION
**Pull request by tiberiuichim**
This is one of the final steps in the addons story: customization capability.

We've enhanced Volto with the following logic:

- Volto projects can now also customize addons, in a `src/customizations/${addon-name}` folder.
- To avoid a mishmash of files in the `src/customizations` folder, all existing Volto customizations can be moved to a folder called `src/customizations/volto`. In any case, multiple customization folders can be specified in the package.json, so it would be possible to dedicate a folder for the "old" customizations style and a new folder for the "addons" customization style.
- Addon packages can now provide `customizationPaths` in their package.json, and they follow the exact pattern as a Volto project.
- The resolve aliases (customizations) are loaded in this order (and last one always wins): customizations coming from addons (in order declared in the project's package.json) and then the project's customizations.

To summarize, addons can customize Volto and other addons, the project comes last and can override any customization and can customize the addons and Volto itself.

### CODE REVIEW

**Comment by tiberiuichim** [comment]
There is a single test missing and for me to write a documentation file, then the PR is ready to be merged.

**Comment by tiberiuichim** [comment]
Fun fact (and very exciting for me): this PR enables complete "decoupling", an addon can do almost everything the project can do. The project can be just an empty "shell" created from create-volto-app and just specify the `addons` in package.json.

**Comment by tiberiuichim** [comment]
This is ready to be merged.

**Review by avoinea — APPROVED** [review]
LGTM.

**Comment by tisto** [comment]
Note volto team: @giuliaghisini or @nzambello will look into it in the next day. @tisto will then merge and release.

**Review by giuliaghisini — APPROVED** [review]

**Review by nzambello — APPROVED** [review]

**Review by tisto — APPROVED** [review]
